junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -51,31 +51,35 @@ private Assertions() {
 	/**
 	 * <em>Fails</em> a test with the given failure {@code message}.
 	 */
-	public static void fail(String message) {
+	public static <V> V fail(String message) {
```

Comment:
No. That won't work. The (**never**) returned type is unknown until compilation. And it has to adopt to every possible type.

`boolean` here: https://github.com/forax/pro/commit/eb5f1b7b6df63c75ffa7a14a61d8d4c596b627bc#diff-3cb01c85a2fef586b2a43f61d42e2f3aR97
`String` there: https://github.com/forax/pro/commit/eb5f1b7b6df63c75ffa7a14a61d8d4c596b627bc#diff-3cb01c85a2fef586b2a43f61d42e2f3aR107